src/python/pants/engine/build_files.py

```diff
@@ -253,32 +262,42 @@ def _hydrate(item_type, spec_path, **kwargs):
 
   # NB: This may be empty, as the result of filtering by tag and exclude patterns!
   return ProvenancedBuildFileAddresses(
-    tuple(
-      ProvenancedBuildFileAddress(
-        build_file_address=addr, provenance=addr_to_provenance[addr]
-      )
-      for addr in matched_addresses
-    )
+    ProvenancedBuildFileAddress(build_file_address=addr, provenance=addr_to_provenance[addr])
+    for addr in matched_addresses
```

Comment:
The `tuple` is not necessary anymore for `engine.objects.Collection[T]` types, as the constructor takes an iterable so works with generator expressions.